.github/workflows/docker.yml

```diff
@@ -90,3 +92,32 @@ jobs:
           file: docker/${{ matrix.dockerfile }}.Dockerfile
           build-args: "SYTEST_IMAGE_TAG=${{ matrix.sytest_image_tag }}"
           tags: ${{ matrix.tags }}
+
+  build-synapse3_10-image:
+    needs: build-sytest-images
+    name: "Build sytest-synapse:bookworm"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Inspect builder
+        run: docker buildx inspect
+
+      - name: Log in to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+      - name: Build and push 3.10
+        uses: docker/build-push-action@v2
+        with:
+          pull: true
+          push: true
+          labels: "gitsha1=${{ github.sha }}"
+          file: docker/synapse3_10.Dockerfile
+          build-args: "SYTEST_IMAGE_TAG=bookworm"
+          tags: "matrixdotorg/sytest-synapse:bookworm"
```

Comment:
I think you can do this more easily by adding an entry to the `matrix` for the `build-dependent-images` job ?